### PR TITLE
Add setMapping to ATP scripting interface and fix threading bug

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 1]
```diff
@@ -143,6 +143,7 @@ ScriptEngine::ScriptEngine(const QString& scriptContents, const QString& fileNam
     _fileNameString(fileNameString),
     _arrayBufferClass(new ArrayBufferClass(this))
 {
+    _assetScriptingInterface.moveToThread(thread());
```
Comment (huffman) [inline libraries/script-engine/src/ScriptEngine.cpp]: The AssetScriptingInterface uses callbacks that we want to get called on the ScriptEngine thread, so move the scripting interface here.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7d491c2b1e7d191f3ed3cfacc88df31a8a9fefa0 available at: https://gist.github.com/e0286b6240469b6b1b3ba4f5bcc7402a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8729-7d491c2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8729-7d491c2.dmg

**Comment by huffman** [comment]
You will need to know how to script to test this PR.

To test:

 1. Use `Assets.uploadData` to upload a short script to the ATP server and get a hash back
   1. You can upload a script like: `print("In the uploaded script!")` - you will run it later
 1. Use `Assets.setMapping(path, hash, callback)` to set an ATP path to point to your new asset.
   1. The path you use will need to start with a `/`
 1. In the script console, use `Script.include("atp:THE_PATH_YOU_USED")` (ex: `Script.include("atp:/test.js")`)
 1. Confirm that the script you uploaded ran

File: libraries/script-engine/src/ScriptEngine.cpp (hunk 1 above)
Comment (Atlante45) [inline libraries/script-engine/src/ScriptEngine.cpp]: `_assetScriptingInterface` has just been created, it is already on `thread()`

Comment (huffman) [inline libraries/script-engine/src/ScriptEngine.cpp]: That has not been my experience, but I will look further

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2a18c7a297ed1af3553fa727abc0aaf6a440a522 available at: https://gist.github.com/e0286b6240469b6b1b3ba4f5bcc7402a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8729-2a18c7a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8729-2a18c7a.dmg

**Comment by thoys** [comment]
## QA Report

```
var ATP_PROTOCOL = 'atp:';
var scriptPath = '/testScript.js';
var scriptContents = 'print("hello world");';
Assets.uploadData(scriptContents, function(url, hash) {
    Assets.setMapping(scriptPath, hash, function() {
        print('uploaded. Attempting to run:');
        Script.include(ATP_PROTOCOL + scriptPath);
    });
});
```
image

My test script worked:

QA Pass

**Pull request merged**